src/natcap/invest/ui_server.py

```diff
@@ -102,6 +104,63 @@ def get_invest_validate():
     return json.dumps(results)
 
 
+@app.route('/colnames', methods=['POST'])
+def get_vector_colnames():
+    """Get a list of column names from a vector.
+    This is used to fill in dropdown menu options in a couple models.
+
+    Body (JSON string):
+        vector_path (string): path to a vector file
+        
+    Returns:
+        a JSON string.
+    """
+    payload = request.get_json()
+    LOGGER.debug(payload)
+    vector_path = payload['vector_path']
+    colnames = []  
+    # a lot of times the path will be empty so don't even try to open it
+    if vector_path:  
+        try:
+            vector = gdal.OpenEx(vector_path, gdal.OF_VECTOR)
+            colnames = [defn.GetName() for defn in vector.GetLayer().schema]
+        except Exception as e:
+            LOGGER.error(f'Could not read column names from {vector_path}. '
+                         f'ERROR: {e}')
```

Comment:
Could I suggest that we change the `LOGGER.error` to `LOGGER.exception`?  As it stands, `f'ERROR: {e}'` will only capture the text of the exception, which with GDAL isn't usually all that helpful without the traceback (which `LOGGER.exception` does capture):

```
>>> try:
...     raster = gdal.Open('foo.tif')
...     band = raster.GetRasterBand(1)
... except Exception as e:
...     print(f'{e}')
...
'NoneType' object has no attribute 'GetRasterBand'
```

Reply:
Sounds good!